### PR TITLE
boy scout rule

### PR DESCRIPTION
**Pull request by eclayaz**

### CODE REVIEW

File: 6-boy-scout-rule/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+###Boy Scout Rule
```
Comment (gayansanjeewa) [inline 6-boy-scout-rule/README.md]:
```suggestion
# Boy Scout Rule
```

File: 6-boy-scout-rule/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    Always leave the code you're editing better than you found it. 
+    ~ Robert C. Martin (Uncle Bob)
+
+![](../images/boy-scout-rule.jpg)
+
+### Speaker notes
```
Comment (gayansanjeewa) [inline 6-boy-scout-rule/README.md:7]: IMO better you create a separate doc for Speaker notes, so as we discussed we use the second monitor to refer to.  Come to think about it I should also create some.

Can you share a doc amongst ourselves?

But do keep some abstract here as well so the audience would see it

File: 6-boy-scout-rule/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+- Boy Scouts have a rule regarding camping, that they should leave the campground cleaner than they found it.
+- Some teams take the approach of shutting down all value-adding work and simply trying to clean up the codebase for a week or a month or some other period of time.
+- The Boy Scout Rule suggests an alternative approach, which is to simply try and ensure that with each commit, you leave the code better than you found it. Eventually you will have a better code.
```
Comment (gayansanjeewa) [inline 6-boy-scout-rule/README.md:10]: Additionally you can add this as well,

* When you find a code that's need to be refactored, but it's not the time for it, set a Todo comment with a counter.
  * `// Todo: Refactor this, requset count 4`
  * And each time some one reach that peace of code and find the need to refactor, they increment that count. When you in the beach and have some time to loose, find the most Todo with the request count and refactor just that

**Review by gayansanjeewa — APPROVED** [review]
LGTM

File: 6-boy-scout-rule/README.md (hunk 2 above)
Comment (eclayaz) [reply 6-boy-scout-rule/README.md:7]: let's discuss about this tomorrow and move to a better place

File: 6-boy-scout-rule/README.md (hunk 3 above)
Comment (eclayaz) [reply 6-boy-scout-rule/README.md:10]: good one

**Pull request merged**